docs/platform-engineers/addon/addon-cue.md

```diff
@@ -0,0 +1,407 @@
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components:
+    - name: namespace
+      type: k8s-objects
+      properties:
+        objects:
+          - apiVersion: v1
+            kind: Namespace
+            metadata:
+              name: my-namespace
+```
+
+> Please notice: Only those CUE files with header `package main` can be reference by `template.cue`, this can be used to help you filter CUE files that you don't want to set into the rendering context.
+
+We just use namespace as example here, other resources of an operator can also be defined in KubeVela application in the same way.
+
+## Features
+
+This section will introduce the way of writing application description file to implement several core features of addon.
+
+### Determine which clusters to be installed by parameters
+
+If you want the resources in the addon to be installed not only in the control-plane, but also in managed clusters, you can use the topology policy in your application as shown below. The parameter `clusters` field will be filled when the addon is enabled with the `clusters` parameter specified.
+
+```cue
+package main
+
+output: {
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
+	}
+	spec: {
+		components:{...}
+		policies: [{
+			type: "topology"
+			name: "deploy-topology"
+			properties: {
+				if parameter.clusters != _|_ {
+					clusters: parameter.clusters
+				}
+				if parameter.clusters == _|_ {
+					clusterLabelSelector: {}
+				}
+			}
+		}]
+	}}
+```
+
+If you execute the command to enable the addon as follows:
+
+```shell
+$ vela addon enable <addon-name> clusters=local,cluser1
+```
+
+The rendering result will be:
+
+```yaml
+kind: Application
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components: ...
+  policies:
+    - type: "topology"
+      name: "deploy-topology"
+      properties:
+        clusters:
+          - local
+          - cluster1
+```
+
+After enabling the addon, the KubeVela controller will install components to the `local` and `cluster1` clusters as defined in the application's topology policy.
+
+If you need to enable the addon in all clusters, you can enable the addon by not setting the `cluster` parameter as follows:
+
+```shell
+$ vela addon enable <addon-name>
+```
+
+The rendering result is :
+
+```yaml
+kind: Application
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components: ...
+  policies:
+    - type: "topology"
+      name: "deploy-topology"
+      properties:
+        clusterLabelSelector: {}
+```
+
+Since an empty (`{}`) `clusterLabelSelector` topology will choose all exist clusters as target, so the components in application will be dispatched to all clusters including control-plane and managed-cluster.
```

Comment:
as target, the components in the application will be dispatched to all clustering including both the control plane and the managed clusters.